subsys/bluetooth/controller/hci/hci.c

```diff
@@ -1502,6 +1502,11 @@ static void le_set_scan_enable(struct net_buf *buf, struct net_buf **evt)
 	status = ll_scan_enable(cmd->enable);
 #endif /* !CONFIG_BT_CTLR_ADV_EXT */
 
+	if (!IS_ENABLED(CONFIG_BT_CTLR_SCAN_ENABLE_STRICT) &&
+	    (status == BT_HCI_ERR_CMD_DISALLOWED)) {
```

Comment:
Is it certain that the only reason `ll_scan_enable()` might return CMD_DISALLOWED is that the current state was already what was requested as the target state?

Reply:
yes. By implementation, Command Disallowed is only returned for failure to start/stop and already started/stopped scanning state. Only other error code returned by ll_scan_enable function is Invalid HCI Command Parameters.